### DISCUSSION TITLE
请问这个工具跟codeql比有什么优势吗？

### DISCUSSION THREAD

**Discussion by kkkOkkkl** [tabby-sec/tabby#49] [General]
请问这个工具跟codeql比有什么优势吗？

**Comment by wh1t3p1g** [comment]
参考 issue #8 
1. 侧重点不一样，codeql 更多面向的是源码级的，tabby 可以直接对编译后的文件做分析
2. 在规则上，neo4j 的语法比较简单易学，是相对固定可以套模版的语法
3. 另外，就是 tabby 最初的实现是用来挖利用链的，目前跟其他工具比，应该是相对好用的挖利用链的工具

**Reply by kkkOkkkl** [reply]
大佬后续的开发计划是做成个国产版的codeql吗？codeql的查询语言作为一个DSL比neo4j强大太多，而且codeql基本上把同类型工具的功能做到尽头了，比较直观的增长点是提升Java指针分析之类的精度，但是这个精度的提升估计很难在短时间内做出来

**Reply by wh1t3p1g** [reply]
目前的计划大概率不会做成 codeql 的模式，DSL 有利有弊吧，把检索算法部分功能开放给用户了，但相对应的这种模式跟 neo4j 语法相比入门门槛会高很多。我目前是实现了 neo4j 的 procedure 来降低这部分的使用门槛。精度问题感觉只能说怎么平衡分析成本和效果这两块。
